model/pdata/common.go

```diff
@@ -116,6 +120,11 @@ func NewAttributeValueArray() AttributeValue {
 	return AttributeValue{orig: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_ArrayValue{ArrayValue: &otlpcommon.ArrayValue{}}}}
 }
 
+// NewAttributeValueBytes creates a new AttributeValue with the given []byte value.
+func NewAttributeValueBytes(v []byte) AttributeValue {
+	return AttributeValue{orig: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_BytesValue{BytesValue: v}}}
+}
```

Comment:
Add a warning about not copying the data, and the data must not be changed after this method is called, or shared between different attributes.

Reply:
We try to not allow users to set/get mutable data structures unless controlled by us.

Reply:
I agree with a comment. Note that this is not unique in any way, it is just another reference data type, like the array or the map that we already allow as attribute value.

Reply:
Right I guess the difference there is with map/array, the NewAttributeValue methods instantiate an empty array/map instead. I'll add a warning.

Reply:
Comment added.